client/client.go

```diff
@@ -37,6 +37,11 @@ type GenericResponse struct {
 // with specific configuration
 func NewClient(clientConfig Configuration) (*gsclientgen.DefaultApi, error) {
 	configuration := gsclientgen.NewConfiguration()
+
+	if clientConfig.Endpoint == "" {
+		return &gsclientgen.DefaultApi{}, microerror.Mask(endpointNotSpecifiedError)
+	}
```

Comment:
We ususally use

```
var invalidConfigError = microerror.New("invalid config")

// IsInvalidConfig asserts invalidConfigError.
func IsInvalidConfig(err error) bool {
	return microerror.Cause(err) == invalidConfigError
}
```
And then

```
return nil, microerror.Maskf(invalidConfigError, "endpoint must not be empty")
```

Reply:
Cool. Learnt something new. That improves things. Can we keep that to a new PR? I basically tried to swap microkit with microerrer (which I understand was a mistake in hindsight)